lib/schema_synchroniser.rb

```diff
@@ -0,0 +1,15 @@
+class SchemaSynchroniser
+  attr_reader :errors
+
+  def initialize(index_group)
+    @index = index_group.current
+  end
+
+  def sync
+    @errors = @index.sync_mappings
+  end
+
+  def synchronised_types
+    @index.mappings.keys.difference(@errors.keys)
+  end
+end
```

Comment:
I considered adding a unit test for this but it ended up basically just testing a mock of the index group, which felt very inflexible. Unfortunately I couldn't see a way to inject a schema into the actual index and index group objects so I decided to stick with the integration test only, but it can't cover the error handling.